src/Microsoft.OData.Core/ODataInnerError.cs

```diff
@@ -93,5 +133,34 @@ internal string ToJson()
                 this.StackTrace == null ? "" : JsonValueUtils.GetEscapedJsonString(this.StackTrace),
                 this.InnerError == null ? "{}" : this.InnerError.ToJson());
         }
-}
+
+        private string LookUpStringValue(string v)
+        {
+            if (Properties == null)
+            {
+                return string.Empty;
+            }
+
+            foreach (KeyValuePair<string, ODataValue> pair in Properties)
+            {
+                if (pair.Key == v)
+                {
+                    string value = (string)pair.Value.FromODataValue();
+                    return value;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private void SetStringValue(string v, string value)
+        {
+            if (Properties == null)
+            {
+                Properties = new Dictionary<string, ODataValue>();
+            }

```

Comment:
Properties should never be null? maybe just Debug.Assert(...)

Reply:
What if default constructor is used?

---
In reply to: [320875481](https://github.com/OData/odata.net/pull/1523#discussion_r320875481) [](ancestors = 320875481)